### PR TITLE
file upload : Handle heavy files

### PR DESCRIPTION
**Pull request by vied12**
related to https://dev.sourcefabric.org/browse/LBSD-283

### CODE REVIEW

**Comment by ioanpocol** [comment]
Looks ok, but it will help to add a test to prove that the large size issue is fixed.

File: superdesk/upload.py [hunk 1]
```diff
@@ -84,23 +94,20 @@ def on_create(self, docs):
...
     def store_file(self, doc, content, filename, content_type):
-        res = process_file_from_stream(content, filename=filename, content_type=content_type)
-        file_name, content_type, metadata = res
-
         cropping_data = self.get_cropping_data(doc)
-        _, out = crop_image(content, filename, cropping_data)
-        metadata['length'] = json.dumps(len(out.getvalue()))
-
+        # crop the file if needed, can change the image size
+        was_cropped, out = crop_image(content, filename, cropping_data)
```
Comment (amagdas) [inline superdesk/upload.py]: Cropping the image first will remove the metadata from the cropped image, that's why we first retrieved the metadata and then cropped the image if needed.

**Comment by vied12** [comment]
updated with test and full metadata
/cc @amagdas @ioanpocol

File: features/upload.feature [hunk 2]
```diff
@@ -4,6 +4,14 @@ Feature: Upload
     Scenario: Upload a binary file
         When we upload a file "bike.jpg" to "/upload"
         Then we get a file reference
+        Then we get "bike.jpg" metadata
+        And we get "avatar" renditions
+        And we can delete that file
+
+    @auth
+    Scenario: Upload a heavy binary file
+        When we upload a file "paris.jpg" to "/upload"
```
Comment (amagdas) [inline features/upload.feature]: I don't think that this test is really needed, if we want to keep it then we should remove the test on top of it as they are duplicates.

**Comment by amagdas** [comment]
Looks good :+1:

**Comment by vied12** [comment]
@amagdas There is a difference between light and heavy file. The file type changes, you think it's not necessary to test both of them ?

**Comment by amagdas** [comment]
Yes, we should keep only the heavy file if we suspect issues with it. I would rather keep the light file as the current test suite is quite slow and adding a heavy file will make it even slower. 
IMO we should keep just the light file test, the rest is good. @ioanpocol do you think we really need the heavy file test?

**Comment by amagdas** [comment]
:+1:

**Pull request merged**